app/javascript/components/download/FileDownloadControl.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState, useEffect } from 'react'
+import { fetchBucketFile } from 'lib/scp-api'
+import { bytesToSize } from 'lib/stats'
+
+/** renders a file download control for the given file object */
+export default function FileDownloadControl({ file, bucketName }) {
+  const [remoteImageUrl, setRemoteImageUrl] = useState(null)
+  const fileName = file.upload_file_name
+
+  /** If there is a remote file, try to load it */
+  useEffect(() => {
+    setRemoteImageUrl(null)
+    getBucketLocalUrl(bucketName, fileName).then(setRemoteImageUrl)
+  }, [bucketName, fileName])
+
+  /** get the remote file path */
+  async function getBucketLocalUrl(bucketName, fileName) {
+    const response = await fetchBucketFile(bucketName, fileName)
```

Comment:
`fetchBucketFile` downloads the file, and we're calling this method every time this component is loaded.  which means we're downloading every file, regardless of whether the user clicks on it.  I don't think that's correct